src/pbcontroller.cpp

```diff
@@ -47,37 +47,32 @@ bool PbController::setup_dirs_xdg(const char* env_home)
 {
 	const char* env_xdg_config;
 	const char* env_xdg_data;
-	std::string xdg_config_dir;
-	std::string xdg_data_dir;
+	newsboat::Filepath xdg_config_dir;
+	newsboat::Filepath xdg_data_dir;
 
 	env_xdg_config = ::getenv("XDG_CONFIG_HOME");
 	if (env_xdg_config) {
 		xdg_config_dir = env_xdg_config;
 	} else {
 		xdg_config_dir = env_home;
-		xdg_config_dir.push_back(NEWSBEUTER_PATH_SEP);
-		xdg_config_dir.append(".config");
+		xdg_config_dir.join(".config");
```

Comment:
`join` doesn't modify the object it's called on. Please use `push` here and further down this function.